linalg/hypre.cpp

```diff
@@ -1587,6 +1640,41 @@ void HypreParMatrix::Destroy()
    }
 }
 
+/* job =  0, extract block diagonal of A and scale A into C
+ * job =  1, job 0 + scale b into d
+ * job =  2, use A to scale b only
+ */
+#if MFEM_HYPRE_VERSION >= 21800
+int BlockInverseScale(const HypreParMatrix *A, HypreParMatrix *C,
+                      const Vector *b, HypreParVector *d, int block, int job)
+{
+   if (0 == job || 1 == job)
+   {
+      hypre_ParCSRMatrix *C_hypre;
+      hypre_ParcsrBdiagInvScal(*A, block, &C_hypre);
+      hypre_ParCSRMatrixDropSmallEntries(C_hypre, 1e-15, 1);
```

Comment:
Is this magic number (1e-15) a good choice?

Reply:
Not magic as much as dropping entries O(1e-16) ~ O(machine precision), but I would often see things like 5 * 1e-16 or some other constant, so I chose 1e-15.

Reply:
Should this `1e-15` be an input argument? Would the user ever want to change it, or should this value work in every possible case?

Reply:
This takes the l1-norm of the row and eliminates entries smaller than 1e-15* the l1-norm. I am not familiar with practical cases where that could mess things up, but can add a switch if people think it should be an option. How accurate is the quadrature computed to?